### PR TITLE
Allow to create tarpc server on existing listener

### PR DESCRIPTION
**Pull request by SabatierBoris**
When you want to use systemd.socket or systemfd for allocate the socket (TCP or UNIX) you need to be able to create a server on already existing listener

### CODE REVIEW

**Comment by tikue** [comment]
Thanks, and apologies for the review delay! Can you add unit tests for the new functions? Other than that, looks great.

**Comment by SabatierBoris** [comment]
> Thanks, and apologies for the review delay! Can you add unit tests for the new functions? Other than that, looks great.

Hi, no problem.

I add some tests based on existing TCP and UDS tests.

File: tarpc/src/serde_transport.rs [hunk 1]
```diff
@@ -669,4 +713,25 @@ mod tests {
         assert_matches!(transport.next().await, None);
         Ok(())
     }
+
+    #[cfg(all(unix, feature = "unix"))]
+    #[tokio::test]
+    async fn uds_on_existing_transport() -> io::Result<()> {
+        use super::unix;
+        use super::*;
```
Comment (tikue) [inline tarpc/src/serde_transport.rs:721]: Just curious, what else was needed besides `super::unix`?

Comment (tikue) [reply tarpc/src/serde_transport.rs:721]: Ah, I think the imports on line 565 just need to be updated to include StreamExt and SinkExt. I see another test already imports super::*, so I'll just update this in a separate commit.

**Review by tikue — APPROVED** [review]

**Pull request merged**